scripts/fetch/fetch.states.R

```diff
@@ -4,21 +4,33 @@
 #' 
 library(maps)
 library(sp)
-proj.string <- "+proj=laea +lat_0=45 +lon_0=-100 +x_0=0 +y_0=0 +a=6370997 +b=6370997 +units=m +no_defs"
-to_sp <- function(...){
+to_sp <- function(..., proj.string, within = NULL){
   library(maptools)
   library(maps)
   map <- maps::map(..., fill=TRUE, plot = FALSE)
   IDs <- sapply(strsplit(map$names, ":"), function(x) x[1])
   map.sp <- map2SpatialPolygons(map, IDs=IDs, proj4string=CRS("+proj=longlat +datum=WGS84"))
   map.sp.t <- spTransform(map.sp, CRS(proj.string))
+  
+  if (!is.null(within)){
+    g.i <- rgeos::gIntersects(map.sp.t, within, byid = T) 
+    map.sp.t <- map.sp.t[g.i]
+  }
+  
   return(map.sp.t)
 }
 
+fetch_map_data <- function(..., viz){
```

Comment:
Is this the right name for this function? Maybe it should be combined with to_sp?

Reply:
to_sp is kind of independent of `vizlab` (and knowing about the `viz` object), which is why i kept those two separate

Reply:
Got it. It makes sense in the context of how it will be used. 👍